home-service/src/main/java/com/softserveinc/ita/homeproject/homeservice/service/impl/CooperationContactServiceImpl.java

```diff
@@ -0,0 +1,61 @@
+package com.softserveinc.ita.homeproject.homeservice.service.impl;
+
+import java.util.List;
+
+import com.softserveinc.ita.homeproject.homedata.entity.Contact;
+import com.softserveinc.ita.homeproject.homedata.entity.ContactType;
+import com.softserveinc.ita.homeproject.homedata.entity.Cooperation;
+import com.softserveinc.ita.homeproject.homedata.repository.ContactRepository;
+import com.softserveinc.ita.homeproject.homedata.repository.CooperationRepository;
+import com.softserveinc.ita.homeproject.homeservice.dto.ContactDto;
+import com.softserveinc.ita.homeproject.homeservice.exception.AlreadyExistHomeException;
+import com.softserveinc.ita.homeproject.homeservice.exception.NotFoundHomeException;
+import com.softserveinc.ita.homeproject.homeservice.mapper.ServiceMapper;
+import com.softserveinc.ita.homeproject.homeservice.service.CooperationContactService;
+import org.springframework.stereotype.Service;
+
+
+@Service
+public class CooperationContactServiceImpl extends BaseContactService implements CooperationContactService {
+
+    private final CooperationRepository cooperationRepository;
+
+    public CooperationContactServiceImpl(ContactRepository contactRepository,
+                                         ServiceMapper mapper, CooperationRepository cooperationRepository) {
+        super(contactRepository, mapper);
+        this.cooperationRepository = cooperationRepository;
+    }
+
+    @Override
+    protected void checkAndFillParentEntity(ContactDto contactDto, Contact createContact, Long parentEntityId) {
+        var cooperation = getCooperationById(parentEntityId);
+        if (Boolean.TRUE.equals(contactDto.getMain())) {
+            List<Contact> getAllContactByCoopId = contactRepository
+                .findAllByCooperationIdAndType(parentEntityId, mapper.convert(contactDto.getType(), ContactType.class));
+            getAllContactByCoopId
+                .stream()
+                .filter(Contact::getMain)
+                .findAny()
+                .ifPresent(contact -> {
+                    throw new AlreadyExistHomeException("Cooperation with id "
+                        + parentEntityId + " already has main " + contactDto.getType() + " contact");
+                });
+        }
+
+        createContact.setCooperation(cooperation);
+    }
+
+    private Cooperation getCooperationById(Long cooperationId) {
```

Comment:
We have to use a service with the corresponding domain. Please inject CoooperationService and retrieve the Cooperation from there. Also, cooperation has to be checked not only for existence but via `enabled` param.

Reply:
You mean that i need use similar name method from side CooperationServiseImpl?